src/ppom-controller.ts

```diff
@@ -589,35 +590,13 @@ export class PPOMController extends BaseControllerV2<
         ),
       true,
     );
-    if (response?.status !== 200) {
-      throw new Error(`Failed to fetch file with url: ${url}`);
+    if (response?.status === 304) {
```

Comment:
Minor, but do we need a `let` here or could we do:

`const cached = response?.status === 304;`